### PR TITLE
2794 - API improvements as per our discussion

### LINKED ISSUE #2794: [Compose] - Improve Channels features structure
### Context & proposal

We need to improve our channels features structure, to mimic what we did for Messages.

### Acceptance criteria

To accept this task we need to complete the following items:

- [x] Split up components by respective files.
- [x] Restructure channels componets in the `components` package.

### PR DESCRIPTION
**Pull request by filbabic**
Fixes #2794 

### 🎯 Goal

Cleaning up the API as per our discussions in chat.

#### General
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-team or #compose-chat-sdk-team) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- [x] Changelog is updated with client-facing changes
~~- [ ] New code is covered by unit tests~~
~~- [ ] Comparison screenshots added for visual changes~~
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channels/list/ChannelItem.kt [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+ * @param leadingContent Customizable composable function that represents the leading content of a channel item, usually
+ * the avatar that holds an image of the channel or its members.
+ * @param detailsContent Customizable composable function that represents the center content of a channel item, usually
+ * holding information about its name and the last message.
+ * @param trailingContent Customizable composable function that represents the trailing content of the a channel item,
+ * usually information about the last message and the number of unread messages.
+ */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+public fun DefaultChannelItem(
```
Comment (MarinTolic) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channels/list/ChannelItem.kt:55]: Do we want to keep the naming "Default" prefix for a component that is public and exposes a lot of slots?

This component seems like something heavily user oriented.

Comment (filbabic) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channels/list/ChannelItem.kt:55]: Definitely should be just `ChannelItem`

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channels/list/ChannelItem.kt [hunk 2]
```diff
@@ -0,0 +1,246 @@
+package io.getstream.chat.android.compose.ui.channels.list
```
Comment (MarinTolic) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channels/list/ChannelItem.kt:1]: Leading Composable is called `DefaultChannelItem` in a file named `ChannelItem.kt`.

Might lead to slightly poorer discoverability.

**Review by MarinTolic — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- ~~[x] UI Changes correct (before & after images)~~ **(no UI changes were made)**
- [x] Bugs validated (bugfixes)
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Pull request merged**